node/syncblock.go

```diff
@@ -368,6 +401,26 @@ func (localNode *LocalNode) initSyncing() {
 // StartSyncing starts block syncing from current local ledger until it gets to
 // block height stopHeight with block hash stopHash from given neighbors
 func (localNode *LocalNode) StartSyncing(syncStopHash common.Uint256, syncStopHeight uint32, neighbors []*RemoteNode) (bool, error) {
+	cs := chain.DefaultLedger.Store.(*store.ChainStore)
+	fastSyncHeight := syncStopHeight - config.Parameters.RecentStateCount
+	log.Info("fast sync begin, stopHeight:", fastSyncHeight)
+	if config.Parameters.SyncMode == "fast" && cs.FastSyncRestorable() {
+		fastSyncRootHash, err := cs.GetCurrentBlockStateRoot()
```

Comment:
Shouldn't use current state root. Consider this case: node stop during fast sync, then disable fast sync and restart. Change current state root will cause node to have wrong state without fast sync enabled.